packages/react/src/hooks/common.ts

```diff
@@ -0,0 +1,57 @@
+import { useContext } from 'react'
+import { InterpreterFrom } from 'xstate'
+
+import { Nhost, NhostMachine } from '@nhost/client'
+import { useSelector } from '@xstate/react'
+
+import { NhostReactContext } from '../provider'
+
+export const useNhost = (): Nhost => {
+  const nhost = useContext(NhostReactContext)
+  return nhost
+}
+
+export const useNhostInterpreter = (): InterpreterFrom<NhostMachine> => {
+  const nhost = useContext(NhostReactContext)
+  if (!nhost.interpreter) throw Error('No interpreter')
+  return nhost.interpreter
+}
+
+export const useNhostBackendUrl = () => {
+  const nhost = useContext(NhostReactContext)
+  return nhost.backendUrl
+}
+
+export const useAuthLoading = () => {
+  const service = useNhostInterpreter()
+  return useSelector(service, (state) => !state.hasTag('ready'))
+}
+
+export const useAuthenticated = () => {
+  const service = useNhostInterpreter()
+  return useSelector(service, (state) => state.matches({ authentication: 'signedIn' }))
+}
```

Comment:
Will this hook automatically set a new value if I'm signed out and then I successfully do `nhost.auth.signIn({email, password});`?

Reply:
yes